tests/unit/errors.js

```diff
@@ -3,19 +3,23 @@ const assert = require('assert');
 const errorsJSON = require('../../errors/arsenalErrors.json');
 const errors = require('../../index').errors;
 
-describe('Runtime errors instance generation', () => {
+describe('Errors: ', () => {
     Object.keys(errors).forEach(index => {
         it(`should return and instance of ${index} Error`, done => {
-            assert.deepStrictEqual(errors[index] instanceof Error, true,
-                                   'should be an instance of Error');
-            assert.deepStrictEqual(errors[index].code, errorsJSON[index].code,
-                                   'Wrong Code');
-            assert.deepStrictEqual(errors[index].description,
-                                   errorsJSON[index].description,
-                                   'Incorrect description');
-            assert.deepStrictEqual(errors[index][index], true,
-                                   `${index} key must be true`);
+            assert.strictEqual(errors[index] instanceof Error, true,
+                'should be an instance of Error');
+            assert.strictEqual(errors[index].code, errorsJSON[index].code,
+                'Wrong Code');
+            assert.strictEqual(errors[index].description,
+                errorsJSON[index].description, 'Incorrect description');
+            assert.strictEqual(errors[index][index], true,
+                `${index} key must be true`);
             done();
         });
     });
+
+    it('should allow custom error descriptions', () => {
+        const error = errors.NoSuchEntity.generateCustom('custom-description');
+        assert.strictEqual(error.description, 'custom-description');
```

Comment:
Let's add another assertion which checks `errors.NoSuchEnitity`'s description is unchanged.